# 0.1. Import Libraries and Data

In [1]:
import pandas as pd
import numpy as np
import os

In [3]:
# Import file

df = pd.read_excel(r"C:\Users\ruthf\OneDrive\Documents\Data Immersion\Achievement 6\Project\02 Data\1. Orignal Data\StockX_data.xlsx",
                  index_col=False)

# 0.2. Data Checks

In [4]:
# Check shape
df.shape

(99956, 8)

In [5]:
# Check head
df.head()

Order Date   Brand                                   Sneaker Name  \
0 2017-09-01   Yeezy           Adidas-Yeezy-Boost-350-Low-V2-Beluga   
1 2017-09-01   Yeezy    Adidas-Yeezy-Boost-350-V2-Core-Black-Copper   
2 2017-09-01   Yeezy     Adidas-Yeezy-Boost-350-V2-Core-Black-Green   
3 2017-09-01   Yeezy       Adidas-Yeezy-Boost-350-V2-Core-Black-Red   
4 2017-09-01   Yeezy  Adidas-Yeezy-Boost-350-V2-Core-Black-Red-2017   

   Sale Price  Retail Price Release Date  Shoe Size  Buyer Region  
0      1097.0           220   2016-09-24       11.0    California  
1       685.0           220   2016-11-23       11.0    California  
2       690.0           220   2016-11-23       11.0    California  
3      1075.0           220   2016-11-23       11.5      Kentucky  
4       828.0           220   2017-02-11       11.0  Rhode Island

In [6]:
# Check tail
df.tail()

Order Date   Brand                                 Sneaker Name  \
99951 2019-02-13   Yeezy  adidas-Yeezy-Boost-350-V2-Static-Reflective   
99952 2019-02-13   Yeezy  adidas-Yeezy-Boost-350-V2-Static-Reflective   
99953 2019-02-13   Yeezy  adidas-Yeezy-Boost-350-V2-Static-Reflective   
99954 2019-02-13   Yeezy  adidas-Yeezy-Boost-350-V2-Static-Reflective   
99955 2019-02-13   Yeezy  adidas-Yeezy-Boost-350-V2-Static-Reflective   

       Sale Price  Retail Price Release Date  Shoe Size Buyer Region  
99951       565.0           220   2018-12-26        8.0       Oregon  
99952       598.0           220   2018-12-26        8.5   California  
99953       605.0           220   2018-12-26        5.5     New York  
99954       650.0           220   2018-12-26       11.0   California  
99955       640.0           220   2018-12-26       11.5        Texas

In [7]:
# Check dtypes
df.dtypes

Order Date      datetime64[ns]
Brand                   object
Sneaker Name            object
Sale Price             float64
Retail Price             int64
Release Date    datetime64[ns]
Shoe Size              float64
Buyer Region            object
dtype: object

In [8]:
# Check for mixed data types
for col in df.columns.tolist():
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df[weird]) > 0:
    print (col)
    

In [9]:
# Check for missing values
df.isnull().sum()

Order Date      0
Brand           0
Sneaker Name    0
Sale Price      0
Retail Price    0
Release Date    0
Shoe Size       0
Buyer Region    0
dtype: int64

In [10]:
# Check for full duplicates
df[df.duplicated()]

Order Date      Brand                           Sneaker Name  \
994   2017-10-13      Yeezy  Adidas-Yeezy-Boost-350-V2-Cream-White   
1028  2017-10-13      Yeezy        Adidas-Yeezy-Boost-350-V2-Zebra   
1513  2017-11-09  Off-White              Nike-Blazer-Mid-Off-White   
1531  2017-11-10      Yeezy  Adidas-Yeezy-Boost-350-V2-Cream-White   
1619  2017-11-11  Off-White              Nike-Air-Presto-Off-White   
...          ...        ...                                    ...   
99815 2019-02-13  Off-White   Nike-Air-Max-90-Off-White-Desert-Ore   
99882 2019-02-13      Yeezy       adidas-Yeezy-Boost-350-V2-Butter   
99907 2019-02-13      Yeezy       adidas-Yeezy-Boost-350-V2-Static   
99924 2019-02-13      Yeezy       adidas-Yeezy-Boost-350-V2-Static   
99926 2019-02-13      Yeezy       adidas-Yeezy-Boost-350-V2-Static   

       Sale Price  Retail Price Release Date  Shoe Size Buyer Region  
994         425.0           220   2017-04-29       10.5     Illinois  
1028        560.0           220   2017-02-25        8.5   California  
1513        550.0           130   2017-09-09        9.0   California  
1531        460.0           220   2017-04-29       10.0   California  
1619        975.0           160   2017-09-09       10.0   California  
...           ...           ...          ...        ...          ...  
99815       520.0           160   2019-02-07        9.0       Oregon  
99882       228.0           220   2018-06-30        9.0       Oregon  
99907       350.0           220   2018-12-27        9.5       Oregon  
99924       330.0           220   2018-12-27        9.0       Oregon  
99926       330.0           220   2018-12-27       10.0       Oregon  

[2840 rows x 8 columns]

#### Although this appears as 2840 duplicates, it is difficult to be 100% sure of this as we have no unique transaction identifier.  Given the size of StockX business I do believe it's possible that the same shoe will sell for the same amount to the same state multiple times within a day.  Unfortunately, this can not be verified.  The data source states: 'every row in the spreadsheet represents an individual StockX sale'.

In [11]:
# Check descriptive stats
df.describe()

Sale Price  Retail Price     Shoe Size
count  99956.000000   99956.00000  99956.000000
mean     446.634754     208.61359      9.344181
std      255.982918      25.20001      2.329588
min      186.000000     130.00000      3.500000
25%      275.000000     220.00000      8.000000
50%      370.000000     220.00000      9.500000
75%      540.000000     220.00000     11.000000
max     4050.000000     250.00000     17.000000

In [12]:
# Check max sale price
df[df['Sale Price']==4050]

Order Date      Brand                               Sneaker Name  \
26870 2018-05-26  Off-White  Air-Jordan-1-Retro-High-Off-White-Chicago   

       Sale Price  Retail Price Release Date  Shoe Size Buyer Region  
26870      4050.0           190   2017-09-09        6.0    Wisconsin

#### No mixed data types or missing data was found.  
#### Descriptive statistics seem fine.  Max price was investigated and it did sell for 4050, the current lowest price is 55,504!!
